Source/Core/VideoCommon/BPMemory.h

```diff
@@ -302,41 +302,37 @@ struct TevStageCombiner
 {
 	union ColorCombiner
 	{
-		struct  //abc=8bit,d=10bit
-		{
-			u32 d : 4; // TEVSELCC_X
-			u32 c : 4; // TEVSELCC_X
-			u32 b : 4; // TEVSELCC_X
-			u32 a : 4; // TEVSELCC_X
+		//abc=8bit,d=10bit
+		BitField<0,  4, u32> d; // TEVSELCC_X
+		BitField<4,  4, u32> c; // TEVSELCC_X
+		BitField<8,  4, u32> b; // TEVSELCC_X
+		BitField<12, 4, u32> a; // TEVSELCC_X
 
-			u32 bias : 2;
-			u32 op : 1;
-			u32 clamp : 1;
+		BitField<16, 2, u32> bias;
+		BitField<18, 1, u32> op;
+		BitField<19, 1, u32> clamp;
 
-			u32 shift : 2;
-			u32 dest : 2;  //1,2,3
+		BitField<20, 2, u32> shift;
+		BitField<22, 2, u32> dest; //1,2,3
 
-		};
 		u32 hex;
 	};
 	union AlphaCombiner
 	{
-		struct
-		{
-			u32 rswap : 2;
-			u32 tswap : 2;
-			u32 d : 3; // TEVSELCA_
-			u32 c : 3; // TEVSELCA_
-			u32 b : 3; // TEVSELCA_
-			u32 a : 3; // TEVSELCA_
-
-			u32 bias : 2; //GXTevBias
-			u32 op : 1;
-			u32 clamp : 1;
-
-			u32 shift : 2;
-			u32 dest : 2;  //1,2,3
-		};
+		BitField<0,  2, u32> rswap;
+		BitField<2,  2, u32> tswap;
```

Comment:
This needs some (int) casts in https://github.com/phire/dolphin/blob/ubershaders/Source/Core/VideoCommon/BPStructs.cpp#L1283